src/shell/autoexec.cpp

```diff
@@ -0,0 +1,675 @@
+/*
+ *  Copyright (C) 2020-2023  The DOSBox Staging Team
+ *  Copyright (C) 2002-2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "autoexec.h"
+
+#include "checks.h"
+#include "control.h"
+#include "dosbox.h"
+#include "fs_utils.h"
+#include "setup.h"
+#include "shell.h"
+#include "string_utils.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+CHECK_NARROWING();
+
+// ***************************************************************************
+// Constants
+// ***************************************************************************
+
+static const std::string autoexec_file_name = "AUTOEXEC.BAT";
+
+constexpr char code_lf = 0x0a; // line feed
+constexpr char code_cr = 0x0d; // carriage return
+
+// ***************************************************************************
+// AUTOEXEC.BAT data - both source and binary
+// ***************************************************************************
+
+using Source = AutoExecObject::Source;
+
+using AutoExecList = std::list<AutoExecObject*>;
+static std::map<Source, AutoExecList> autoexec_objects;
+
+static std::string autoexec_blob_utf8 = {};
```

Comment:
blob, data... I always try to avoid these names as they can mean pretty much anything 😄 What does this particular blob represent? 😄  Can we call it that instead?

Reply:
Renamed to `autoexec_bat_utf8`.